slometricsemitter/log_processor.go

```diff
@@ -104,6 +211,93 @@ func (sloMetricsProc *sloMetricsProcessor) setupTelemetry() error {
 	return nil
 }
 
+
+func (sloMetricsProc *sloMetricsProcessor) startPeriodicRefreshLogic(ctx context.Context, bucketName, httpLocalPath, rpc2LocalPath string) {
+
+	go func() {
+		ticker := time.NewTicker(1 * time.Hour)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.C:
+
+				// Fetch latest SLO configs from S3
+				err := sloMetricsProc.FetchSLOConfigFromS3(ctx, bucketName, httpLocalPath, rpc2LocalPath)
+				if err != nil {
+					sloMetricsProc.logger.Error("Failed to refresh SLO config from S3", zap.Error(err))
+					continue
+				}
+				
+			}
+		}
+	}()
+}
+
+func (sloMetricsProc *sloMetricsProcessor) reloadAndMergeSLOConfig() error {
+
+	httpLocalPath := "./http.yaml"
+	rpc2LocalPath := "./rpc2.yaml"
+
+	// Read HTTP SLO config
+	httpConfig, err := sloMetricsProc.readSLOConfigFile(httpLocalPath)
+	if err != nil {
+		sloMetricsProc.logger.Error("Failed to reload HTTP SLO config file", zap.String("path", httpLocalPath), zap.Error(err))
+		return err
+	}
+
+	// Read RPC2 SLO config
+	rpc2Config, err := sloMetricsProc.readSLOConfigFile(rpc2LocalPath)
+	if err != nil {
+		sloMetricsProc.logger.Error("Failed to reload RPC2 SLO config file", zap.String("path", rpc2LocalPath), zap.Error(err))
+		return err
+	}
+
+	// Merge both configurations safely
+	sloMetricsProc.Lock()
+	defer sloMetricsProc.Unlock()
+
+	// If sloConfig is identical, avoid redundant updates
+	if reflect.DeepEqual(sloMetricsProc.sloConfig, sloMetricsProc.mergeSLOConfigs(httpConfig, rpc2Config)) {
+		sloMetricsProc.logger.Error("No changes detected, skipping config update.")
+		return nil
+	}
+
+	sloMetricsProc.sloConfig = sloMetricsProc.mergeSLOConfigs(httpConfig, rpc2Config)
+
+	return nil
+}
+
+
+func (sloMetricsProc *sloMetricsProcessor) mergeSLOConfigs(httpConfig, rpc2Config SLOConfig) SLOConfig {
+
+    // Create a new SLOConfig to hold the merged result
+    merged := SLOConfig{
+		SLOs: make(map[string]map[string]EndpointSLOConfig), // Initialize the top-level map for SLOs
+    }
+
+    // Initialize the "http" section if it exists in the httpConfig
+    if httpConfig.SLOs["http"] != nil {
+		merged.SLOs["http"] = make(map[string]EndpointSLOConfig)
+        for key, value := range httpConfig.SLOs["http"] {
+            merged.SLOs["http"][key] = value
+        }
+    }
+
+    // Initialize the "rpc2" section if it exists in the rpc2Config
+    if rpc2Config.SLOs["rpc2"] != nil {
```

Comment:
the rpc.yaml file will contain both `rpc2:` and `grpc:` sections. we need to update so that grpc endpoints will look up using the rpc.yaml file but within the `grpc` key